src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlEventSourceListener.netfx.cs

```diff
@@ -144,7 +144,12 @@ private void OnBeginExecute(EventWrittenEventArgs eventData)
                 this.options.AddConnectionLevelDetailsToActivity((string)eventData.Payload[1], activity);
 
                 string commandText = (string)eventData.Payload[3];
-                if (!string.IsNullOrEmpty(commandText) && this.options.SetDbStatement)
+
+                // When using the adoNetEvent source we can only capture the statement text for CommandType.StoredProcedure anyway,
+                // so we check for `SetDbStatementForStoredProcedure`.
+                // Otherwise we cannot distinguish between CommandType.Text and CommandType.StoredProcedure, so we only set the tag if `SetDbStatementForText` is true.
+                if (!string.IsNullOrEmpty(commandText) &&
+                    (this.options.SetDbStatementForText || (eventData.EventSource == this.adoNetEventSource && this.options.SetDbStatementForStoredProcedure)))
```

Comment:
Should this be something like:
```csharp
bool captureText = this.options.SetDbStatementForText && eventData.EventSource == this.mdsEventSource;
bool captureSp = this.options.SetDbStatementForStoredProcedure && eventData.EventSource == this.adoNetEventSource;

if (captureText || captureSp) { ... }
```

Reply:
The logic looks good to me, but it could lead to some confusion. If you have `SetDbStatementForText=true` & `SetDbStatementForStoredProcedure=false` and `eventData.EventSource != this.adoNetEventSource` then you could still end up with stored procedures, which you told it you didn't want. Right?

That is probably not the end of the world, but maybe it would be more clear as an enum?

```csharp
public enum SqlCommandTextCaptureMode
{
   None,
   StoredProcedure,
   StoredProducureAndText
}
```

🤷

Reply:
But then you couldn't capture only Text which is currently possibly under .net core, ok you could turn the enum into flags... But personally I would leave it as it is/

Reply:
Ya I did that intentionally. My thinking was... having the two isn't particularly useful to anyone. Is there a use case where you would want Text and not Stored Procedures? I doubt it :smile: IMO most people will do Stored Procedures, some will also kick on Text. But Text and NOT Stored Procedures? The enum simplifies it and is trying to avoid letting users configure something that is impossible.

Reply:
StoredProducureAndText is still impossible on .NET framework with System.Data.SqlClient . In any case this will be a breaking change of the API for .NET core users. So far it's only breaking for framework users. As I said, I think the way it is now is fine, but I will change it to an enum if that will get the PR merged :)